### PR TITLE
[WIP] Added caching chapter

### PR DESCRIPTION
**Pull request by dantleech**
Caching

### CODE REVIEW

File: cookbook/caching-with-varnish.rst [hunk 1]
```diff
@@ -0,0 +1,242 @@
+Caching with Varnish
+====================
+
+Varnish is a HTTP `caching proxy server`_ which can be used to radically
+improve the response time of your website.
+
+Sulu is bundled with a "soft" caching proxy, the `Symfony HttpCache`_, however
+this is not as fast or powerful as Varnish.
+
+This tutorial will walk you thorugh the process of setting up Varnish on
```
Comment (danrot) [inline cookbook/caching-with-varnish.rst]: through

File: cookbook/caching-with-varnish.rst [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+====================
+
+Varnish is a HTTP `caching proxy server`_ which can be used to radically
+improve the response time of your website.
+
+Sulu is bundled with a "soft" caching proxy, the `Symfony HttpCache`_, however
+this is not as fast or powerful as Varnish.
+
+This tutorial will walk you thorugh the process of setting up Varnish on 
+your own server and configuring to work with SuluCMF.
```
Comment (danrot) [inline cookbook/caching-with-varnish.rst]: configuring it (?)

File: cookbook/caching-with-varnish.rst [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+    # app/config/config.yml
+    # ...
+    fos_http_cache: 
+        proxy_client: 
+            varnish: 
+                servers: localhost:8083 
+                base_url: <your domain name>
+
+Replace ``<your domain name`` with the corresponding value, e.g. ``www.mydomain.dom``.
```
Comment (danrot) [inline cookbook/caching-with-varnish.rst]: <your domain name>

File: cookbook/caching-with-varnish.rst [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+First tell the ``FOSHttpCacheBundle`` about the new varnish server:
+
+.. code-block:: yaml
+
+    # app/config/config.yml
+    # ...
+    fos_http_cache: 
+        proxy_client: 
+            varnish: 
+                servers: localhost:8083
```
Comment (danrot) [inline cookbook/caching-with-varnish.rst]: Is it 8083?

Comment (dantleech) [inline cookbook/caching-with-varnish.rst]: By defailt it is `8080`, which conflicts with Jackrabbit. I will add a note.

Comment (danrot) [inline cookbook/caching-with-varnish.rst]: But above you used the port `8090`, shouldn't this be the value then?

**Comment by dantleech** [comment]
Updated and tests pass.

**Comment by dantleech** [comment]
Updated.

Need to add bundle documentation / reference.

File: cookbook/caching-with-varnish.rst [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+
+.. code-block:: bash
+
+    $ ps ax | grep varnish
+    6585 ?        SLs    0:00 varnishd -f /home/daniel/.varnish/sulu.vcl -s malloc,1G -T 127.0.0.1:2000 -a 0.0.0.0:6081
+    6609 ?        Sl     0:07 varnishd -f /home/daniel/.varnish/sulu.vcl -s malloc,1G -T 127.0.0.1:2000 -a 0.0.0.0:6081
+
+The ``-a`` option indicates where Varnish is listening - it is listening on port ``8083``, which is incorrect.
+
+Under Debiae/Ubuntu we can change the initialization script:
```
Comment (dantleech) [inline cookbook/caching-with-varnish.rst]: Debian

**Pull request merged**